FrontEnd/src/components/ProfilePage/FormComponents/UserInfo.js

```diff
@@ -52,6 +50,34 @@ const UserInfo = (props) => {
         props.currentFormNameHandler(props.curForm);
     }, []);
 
+    const errorMessageTemplates = {
+        fieldLength: 'Введіть від 2 до 50 символів',
+        notAllowedSymbols: 'Поле містить недопустимі символи та/або цифри',
+      };
+
+    const validateFields = (fieldName, fieldValue) => {
+        const allowedSymbolsPatterns = {
+            'person_position': /^[a-zA-Zа-щюяьА-ЩЮЯЬїЇіІєЄґҐ\-'\s]+$/,
+            'name': /^[a-zA-Zа-щюяьА-ЩЮЯЬїЇіІєЄґҐ'\s]+$/,
+            'surname': /^[a-zA-Zа-щюяьА-ЩЮЯЬїЇіІєЄґҐ'\s]+$/,
+        };
+        const letterCount = (fieldValue.match(/[a-zA-Zа-щюяьА-ЩЮЯЬїЇіІєЄґҐ]/g) || []).length;
+        const isValidLength = letterCount >= 2 || (fieldName === 'person_position' && letterCount === 0);
+        const isValidPattern = allowedSymbolsPatterns[fieldName].test(fieldValue);
+        let errorMessage = '';
+
+        if (fieldValue && !isValidPattern) {
+            errorMessage = errorMessageTemplates.notAllowedSymbols;
+        } else if (!isValidLength) {
+            errorMessage = errorMessageTemplates.fieldLength;
+        }
```

Comment:
If both `!isValidPattern` and `!isValidLength` are true, the latter's message will overwrite the former's, which might not be the intended behavior. Consider handling the case where both conditions could be true and both messages might need to be displayed

Reply:
Updated
![image](https://github.com/ita-social-projects/Forum/assets/114805677/aa2fb647-486c-47d1-80e8-45e730694d00)